Update global tags to use V5 JEC for 2017 data/MC and add V7b JR for 2018 data/MC [10_6_X]

christopheralanwest opened the pull request:
#### PR description:

First of all, please note that this PR does not contain the recently produced updated 2018 JEC. The tags with the 2018 JEC exist and should be uploaded very soon (tomorrow?).

This PR is a backport of #30478, which should have been prepared in July along with the original PR. This backport PR does not update the JEC for Run 3 scenarios to avoid unnecessary changes.

In addition, there is a correction to the tags for 2018 GTs that will be propagated to master in the next few days with the updated 2018 JEC. For example, in `112X_upgrade2018_realistic_v4`, the `JetResolutionRcd` with label `AK8PFchs_phi` currently uses the tag `JR_Autumn18_V7b_MC_PtResolution_AK8PFchs` (a Pt resolution tag) rather than `JR_Autumn18_V7b_MC_PhiResolution_AK8PFchs`.

The change of HLT JEC in the 2017 cosmics GTs is a purely technical change associated with an incorrectly assigned synchronization; the payload hashes for the two tags are identical. Of course, the HLT JEC are not relevant for the cosmic GT in any case.

The GT diffs for this PR are as follows:

**Offline data**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_dataRun2_v29/106X_dataRun2_v30

**Offline data relval**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_dataRun2_relval_v27/106X_dataRun2_relval_v28

**2017 realistic**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_mc2017_realistic_v7/106X_mc2017_realistic_v8

**2017 realistic cosmics (tracker deco mode)**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_mc2017cosmics_realistic_deco_v2/106X_mc2017cosmics_realistic_deco_v3

**2017 realistic cosmics (tracker peak mode)**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_mc2017cosmics_realistic_peak_v2/106X_mc2017cosmics_realistic_peak_v3

**2018 realistic**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018_realistic_v11/106X_upgrade2018_realistic_v12

**2018 realistic (HEM15/16 failure)**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018_realistic_HEfail_v11/106X_upgrade2018_realistic_HEfail_v12

**2018 cosmics (tracker deco mode)**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018cosmics_realistic_deco_v9/106X_upgrade2018cosmics_realistic_deco_v10

**2018 cosmics (tracker peak mode)**
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/106X_upgrade2018cosmics_realistic_peak_v9/106X_upgrade2018cosmics_realistic_peak_v10

#### PR validation:

Please see PR #30478 for details of the validation. In addition, a technical test was performed: `runTheMatrix.py -l limited,7.21,11024.2,7.4 --ibeos`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

This PR is a backport of PR #30478. The global tags contained in this PR are needed for the Run 2 re-miniAOD.

christopheralanwest committed:
Use JEC UL17 V5 for 2017 data and MC and add V7b jet resolution corrections to 2018 data and MC

christopheralanwest commented:
test parameters
workflow = 7.21,11024.2,7.4

cmsbuild commented:
A new Pull Request was created by  @christopheralanwest  for CMSSW_10_6_X.

It involves the following packages:

Configuration/AlCa


@yuanchao, @christopheralanwest, @tocheng, @cmsbuild, @tlampen, @pohsun can you please review it and eventually sign? Thanks.
@makortel, @Martin-Grunewald, @mmusich, @fabiocos, @tocheng this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

christopheralanwest commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
Test Parameters:
  - **MATRIX_EXTRAS** = 7.21,11024.2,7.4

cmsbuild commented:
+1
Tested at: 4d32f94f1769e110d29fbbd574599f4be5042825
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13afa1/10679/summary.html
CMSSW: CMSSW_10_6_X_2020-11-11-2300
SCRAM_ARCH: slc7_amd64_gcc700

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13afa1/10679/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-13afa1/11024.2_TTbar_13UP18HEfailINPUT+TTbar_13UP18HEfailINPUT+DigiFullHEfail+RecoFullHEfail+HARVESTFullHEfail+NanoFullHEfail
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-13afa1/7.21_Cosmics_UP17+Cosmics_UP17+DIGICOS_UP17+RECOCOS_UP17+ALCACOS_UP17+HARVESTCOS_UP17

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2003 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 3214708
 - DQMHistoTests: Total failures: 9046
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3205328
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 140 log files, 14 edm output root files, 34 DQM output files

christopheralanwest commented:
+1

- Updated 2017 UL JEC for data/MC
- Updated 2018 JEC for data/MC to [be consistent with 102X GTs](https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4309/2/1/1/1/1/1/1/1/1/1/1/1/1.html); note that this does _not_ include the UL18 V5 JEC, which will come in a later PR
- changes in 2017 and 2018 data/MC only seen in expected workflows
- pre-2017 changes only seen in slimmed Puppi MET significance (workflows 4.53, 136.731 and 136.7611). This minor change is [expected as a consequence of the extended JetPt range](https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4309/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1.html).

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_10_6_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_11_2_X is complete. This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
